packages/app-builder/src/routes/ressources+/data+/editField.tsx

```diff
@@ -78,11 +85,71 @@ export async function action({ request }: ActionFunctionArgs) {
   }
 }
 
+type fieldIsUniqueOverride = {
+  checked: boolean;
+  disabled: boolean;
+  reason:
+    | 'cannot_toggle_enum_enabled'
+    | 'cannot_toggle_index_pending'
+    | 'cannot_untoggle_field_linked'
+    | 'object_id_must_be_unique'
+    | '';
+};
+
+function overrideStatusFieldUnique({
+  field,
+  linksToThisTable,
+  selectedEnum,
+}: {
+  field: DataModelField;
+  linksToThisTable: LinksToSingle[];
+  selectedEnum: boolean;
+}): fieldIsUniqueOverride {
+  if (field.name === 'object_id') {
+    return {
+      checked: true,
+      disabled: true,
+      reason: 'object_id_must_be_unique',
+    };
+  }
+  if (selectedEnum) {
+    return {
+      checked: false,
+      disabled: true,
+      reason: 'cannot_toggle_enum_enabled',
+    };
+  }
+  if (field.unicityConstraint === 'pending_unique_constraint') {
+    return {
+      checked: true,
+      disabled: true,
+      reason: 'cannot_toggle_index_pending',
+    };
+  }
+  const linksToThisField = linksToThisTable.filter(
+    (link) => link.parentFieldName === field.name,
+  );
+  if (linksToThisField.length > 0) {
+    return {
+      checked: true,
+      disabled: true,
+      reason: 'cannot_untoggle_field_linked',
+    };
+  }
+  return {
+    disabled: false,
+    checked: false,
+    reason: '',
+  };
+}
```

Comment:
IMO, we shouldn't override `checked` property. The value is hold by the data model.

Still, I think disabling (with reason) part of the UI helps prevent unsuitable actions from the user (which in the end will result in errors thrown by the backend) but :
- I prefer to display real value, so the user (and devs) are not confused when there is a problem
- Normally, this kind of "disable stuff we do not want to allow" should comes after handling the possible errors from the backend (but in this special case, I must admit we can skip it, it's long and cumbersome).

Reply:
Ok, I understand your point.
The way I ended up there I think is that I wanted to force-display a checked box if `field.unicityConstraint === 'pending_unique_constraint'`. Probably at first I was returning a value for checked only in this case. But then I was using a possible undefined in the `checked={field.value || fieldIsUniqueOverride.checked}`, and the logic ended up depending on the order of values in the `||` (I think I was getting undefined checkbox values (though still displayed as true I think) in the case `undefined || false`, or maybe the other way around, and it was a headache).

Reply:
Alright perhaps something like 
```
checked={
  fieldIsUniqueOverride.reason ===
  'cannot_toggle_index_pending'
    ? true
    : field.value
}
```
does the job.

Reply:
or nevermind, I guess we don't need to override the value at all indeed